src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/functions.ts

```diff
@@ -93,6 +99,20 @@ export const filterFunctionDefinitions = (
     if (ignoreAsSuggestion) {
       return false;
     }
+    const hasRestrictedSignature = signatures.some((signature) => signature.license);
+    if (!!hasMinimumLicenseRequired && hasRestrictedSignature) {
+      const availableSignatures = signatures.filter((signature) => {
+        if (!signature.license) return true;
+        // signature.license is capitalized, so we need to lowercase it to match the license type passed to the function
```

Comment:
Maybe narrowing license types in signatures (see other comment) would help with this:

```ts
licesnse: 'PLATINUM';
```